### PR TITLE
[cond] support torch built in function as subgraph

### PR DESCRIPTION
**Pull request by ydwu4**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #126909
* #127026

Fixes https://github.com/pytorch/pytorch/issues/126818.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

**Comment by galv** [comment]
thank you!

File: torch/_dynamo/variables/higher_order_ops.py [hunk 1]
```diff
@@ -131,6 +130,28 @@ def _assert_tensors_nonaliasing(inputs, outputs):
...
+        func_var,
+        (
+            UserFunctionVariable,
+            NestedUserFunctionVariable,
+            NNModuleVariable,
+            UnspecializedNNModuleVariable,
+            TorchInGraphFunctionVariable,
+        ),
+    ):
+        unimplemented(f"{arg_name} is of unsupported callable type {str(func_var)}.")
```
Comment (zou3519) [inline torch/_dynamo/variables/higher_order_ops.py:152]: Is it possible to just run `callable(func_var)` ?

**Review by zou3519 — APPROVED** [review]

File: torch/_dynamo/variables/higher_order_ops.py (hunk 1 above)
Comment (ydwu4) [reply torch/_dynamo/variables/higher_order_ops.py:152]: That's a good idea but these dynamo variables don't have a `__call__` method. It iis handled as call_function ("__call__"...). We may test callable for the wrapped actual python object though but it seems requiring a reliable way of getting the inner python object out. We can revisit it in the future.

Comment (zou3519) [reply torch/_dynamo/variables/higher_order_ops.py:152]: torch.compile seems like it supports checking if an object `callable`, so we could figure out how to leverage that

image

Comment (ydwu4) [reply torch/_dynamo/variables/higher_order_ops.py:152]: Oh, yeah, we could do similar things like how pytree.flatten in dynamo. I can change to that.

**Pull request closed without merging**